src/NuGetGallery/Areas/Admin/Controllers/AdminControllerBase.cs

```diff
@@ -1,12 +1,109 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Services.Entities;
+using NuGet.Versioning;
+using NuGetGallery.Areas.Admin.ViewModels;
 using NuGetGallery.Filters;
 
 namespace NuGetGallery.Areas.Admin.Controllers
 {
     [UIAuthorize(Roles="Admins")]
     public class AdminControllerBase : AppController
     {
+        internal List<Package> SearchForPackages(IPackageService packageService, string query)
+        {
+            // Search supports several options:
+            //   1) Full package id (e.g. jQuery)
+            //   2) Full package id + version (e.g. jQuery 1.9.0, jQuery/1.9.0)
+            //   3) Any of the above separated by comma
+            // We are not using Lucene index here as we want to have the database values.
+
+            var queryParts = query.Split(new[] { ',', '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
+
+            var packages = new List<Package>();
+            var completedQueries = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            foreach (var queryPart in queryParts)
+            {
+                var spitQuery = queryPart.Split(new[] { ' ', '/' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
```suggestion
                var splitQuery = queryPart.Split(new[] { ' ', '/' }, StringSplitOptions.RemoveEmptyEntries);
```

(Though `spitQuery does sound very cool!)